openpype/hosts/cinema4d/plugins/publish/collect_instances.py

```diff
@@ -0,0 +1,185 @@
+        ctx_frame_end = context.data['frameEnd']
+        ctx_handle_start = context.data['handleStart']
+        ctx_handle_end = context.data['handleEnd']
+        ctx_frame_start_handle = context.data['frameStartHandle']
+        ctx_frame_end_handle = context.data['frameEndHandle']
+
+        context.data['objectsets'] = objectset
+        for objset in objectset:
+
+            if not objset.get("id"):
+                continue
+
+            if objset.get("id") != "pyblish.avalon.instance":
+                continue
+
+            # The developer is responsible for specifying
+            # the family of each instance.
+            has_family = objset.get("family")
+            assert has_family, "\"%s\" was missing a family" % objset
+
+            members = objset.get("SELECTIONOBJECT_LIST")
+            if members is None:
+                self.log.warning("Skipped empty instance: \"%s\" " % objset)
+                continue
+
+            self.log.info("Creating instance for {}".format(lib.serialize_c4d_data(objset.op)))
+
+            data = dict()
+
+            for key, value in objset:
+                data[key] = lib.serialize_c4d_data(value)
+
+            # temporarily translation of `active` to `publish` till issue has
+            # been resolved, https://github.com/pyblish/pyblish-base/issues/307
+            if "active" in data:
+                data["publish"] = data["active"]
+
+            # Collect members - c4d.BaseObject
+            members = [lib.ObjectPath(obj=members.ObjectFromIndex(doc, idx)) for idx in range(members.GetObjectCount())]
+
+
+            # Collect Children - c4d.BaseObject
+            children = [lib.ObjectPath(obj=obj) for member in members for obj in lib.walk_hierarchy(member.obj.GetDown())]
+
+            parents = []
+            if data.get("includeParentHierarchy", True):
+                # If `includeParentHierarchy` then include the parents
+                # so they will also be picked up in the instance by validators
+                parents = [lib.ObjectPath(obj=obj) for obj in self.get_all_parents(members)]
+            members_hierarchy = list(
+                    set(
+                        [str(x) for x in members] + \
+                        [str(x) for x in children] + \
+                        [str(x) for x in parents]
+                    )
+                )
+
+            if 'families' not in data:
+                data['families'] = [data.get('family')]
+
+            # Create the instance
+            instance = context.create_instance(objset.op.GetName())
+            instance[:] = members_hierarchy
+
+            # Store the exact members of the object set
+            instance.data["setMembers"] = [lib.serialize_c4d_data(x) for x in members]
+
+            # Define nice label
+            name = objset.op.GetName()  # use short name
+            label = "{0} ({1})".format(name,
+                                       data["asset"])
```

Comment:
```suggestion
            label = "{0} ({1})".format(name, data["asset"])
```

Reply:
Thanks for taking a look! Sorry about the volume of code, I did a lot of copying and pasting from Maya/Blender/Nuke hosts to figure out how to put this together so I am sure there are a few missteps, and vestigial code from the other hosts that still needs to be cleaned out. I really appreciate you taking the time!